keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/ArtifactUtils.kt

```diff
@@ -24,18 +24,20 @@ import com.fasterxml.jackson.module.kotlin.readValue
 import com.netflix.spinnaker.exceptions.ArtifactParsingException
 import com.netflix.spinnaker.keel.api.artifacts.ArtifactStatus
 import com.netflix.spinnaker.keel.api.artifacts.ArtifactType
-import com.netflix.spinnaker.keel.api.artifacts.ArtifactType.deb
-import com.netflix.spinnaker.keel.api.artifacts.ArtifactType.docker
-import com.netflix.spinnaker.keel.api.artifacts.DebianArtifact
 import com.netflix.spinnaker.keel.api.artifacts.DeliveryArtifact
-import com.netflix.spinnaker.keel.api.artifacts.DockerArtifact
 import com.netflix.spinnaker.keel.api.artifacts.VirtualMachineOptions
+import com.netflix.spinnaker.keel.artifact.DEB
+import com.netflix.spinnaker.keel.artifact.DOCKER
+import com.netflix.spinnaker.keel.artifact.DebianArtifact
+import com.netflix.spinnaker.keel.artifact.DockerArtifact
 import com.netflix.spinnaker.keel.serialization.configuredObjectMapper
 
 private val objectMapper: ObjectMapper = configuredObjectMapper()
 
 /**
  * A helper function to construct the proper artifact type from the serialized json.
+ * FIXME: this needs to go away in favor of [ArtifactPublisher] functions to keep
+ * artifact contracts generic.
  */
 fun mapToArtifact(
```

Comment:
@emjburns Would love some context on why we needed custom deserialization for artifacts in the first place.

Reply:
IIRC because otherwise we couldn't read the artifacts from the database into their proper types.